src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let stateClone = { ...state };
+  const list = [];
+
+  function removeProperties(obj, keysToRemove) {
+    for (let index = 0; index < keysToRemove.length; index++) {
+      delete stateClone[keysToRemove[index]];
+      delete obj[keysToRemove[index]];
+    }
+
+    return obj;
+  }
 
+  for (let index = 0; index < actions.length; index++) {
+    const item = actions[index];
+
+    switch (item.type) {
+      case 'addProperties':
+        stateClone = { ...stateClone, ...item.extraData };
+        list.push({ ...stateClone, ...item.extraData });
+        break;
+      case 'removeProperties':
+        list.push(removeProperties({ ...stateClone }, item.keysToRemove));
+        break;
+      case 'clear':
+        stateClone = {};
+        list.push({});
```

Comment:
When clearing the state, you are correctly creating an empty object. However, you could optimize this by not assigning '{}' to 'stateClone' since you are immediately pushing an empty object to 'list'. You can simply create a new empty object within the 'list.push' call.